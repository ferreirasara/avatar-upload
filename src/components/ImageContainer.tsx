import { ImageContainerProps } from "../@types/ImageContainer";
import { ReactComponent as InfoIcon } from '../icons/info.svg';

const ImagePlaceholder = () => {
  return <div className="image-container placeholder">
    <InfoIcon />
  </div>
}

export const ImageContainer = (props: ImageContainerProps) => {
  const { image } = props;
  return <div className="image-container">
    {image ? <img src={image} className="image-container img" />
      : <ImagePlaceholder />
    }
  </div>
}