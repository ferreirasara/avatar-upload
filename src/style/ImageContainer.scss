.image-container {
  width: 113px;
  height: 113px;
  border-radius: 100%;
  display: flex;
  justify-content: center;
  align-items: center;

  .placeholder {
    background-color: #C3CBD5;
  }

  .img {
    max-width: auto;
    max-height: auto;
    object-fit: cover;
  }
}

.avatar-uploader-container {
  height: 109px;
  width: 485px;
  border-radius: 8px;
  background-color: #F2F5F8;
  display: flex;
  justify-content: space-between;
  align-items: center;
  text-align: center;
  padding: 32px;
  gap: 32px;

  &.bordered {
    border: 2px dashed #C7CDD3;
  }
}