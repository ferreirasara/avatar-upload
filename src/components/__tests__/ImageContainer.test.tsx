import renderer from 'react-test-renderer';
import { ImageContainer } from '../ImageContainer';

describe('ImageContainer', () => {
  it('renders correctly', () => {
    const imageContainer = renderer.create(<ImageContainer image='https://dummyimage.com/largerectangle' />).toJSON();
    expect(imageContainer).toMatchSnapshot();
  });
})