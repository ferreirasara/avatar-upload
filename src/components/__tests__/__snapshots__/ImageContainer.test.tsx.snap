// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageContainer renders correctly 1`] = `
<div
  className="image-container"
>
  <img
    className="image-container img"
    src="https://dummyimage.com/largerectangle"
  />
</div>
`;
